app/helpers/download_helper.rb

```diff
@@ -0,0 +1,22 @@
+# Custom download links based on how MIT implements geoblacklight
+module DownloadHelper
+  # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Presigner.html
+  def presign_this(document)
+    # for minio
+    if Rails.env.test? || Rails.env.development?
+      Aws.config[:s3] = { endpoint: ENV['MINIO_URL'], force_path_style: true }
+    end
+
+    signer = Aws::S3::Presigner.new
+
+    signer.presigned_url(:get_object, bucket: ENV['AWS_S3_BUCKET'],
+                                      key: key_predictor(document),
+                                      expires_in: 900)
+  end
+
+  # This assumes everything is a zip file. I believe that is our assumption
```

Comment:
Please let me know if this assumption is valid. I made it based on my limited understanding of the systems in play.

Reply:
Yeah, this assumption is correct.